### ISSUE TITLE
Error with code snippet

### ISSUE THREAD

**Issue by grilledham** [liwangqian/LuaCoderAssist#56]
I noticed LuaCoderAssist fails to parse some code. 
This is the simplest code snippet I can make that produces an error.

```lua
local function foo(tbl, index)
    local item = tbl[index] or {amount = 0}
    item.amount = item.amount
end
```
```
[Info  - 11:05:07 PM] coder inited
[Info  - 11:05:07 PM] extension initialize status: true
[Error - 11:05:07 PM] TypeError: Cannot read property 'valid' of undefined
    at LuaSymbol.get valid [as valid] (c:\Users\me\.vscode\extensions\liwangqian.luacoderassist-2.2.7\server\lib\engine\symbol.js:126:27)
    at LuaTable.set (c:\Users\me\.vscode\extensions\liwangqian.luacoderassist-2.2.7\server\lib\engine\symbol.js:307:32)
    at LuaSymbol.set (c:\Users\me\.vscode\extensions\liwangqian.luacoderassist-2.2.7\server\lib\engine\symbol.js:122:19)
    at parseInitStatement (c:\Users\me\.vscode\extensions\liwangqian.luacoderassist-2.2.7\server\lib\engine\core.js:194:26)
    at parseInitStatement (c:\Users\me\.vscode\extensions\liwangqian.luacoderassist-2.2.7\server\lib\engine\core.js:97:13)
    at node.variables.forEach (c:\Users\me\.vscode\extensions\liwangqian.luacoderassist-2.2.7\server\lib\engine\core.js:191:13)
    at Array.forEach (<anonymous>)
    at parseAssignmentStatement (c:\Users\me\.vscode\extensions\liwangqian.luacoderassist-2.2.7\server\lib\engine\core.js:153:24)
    at walkNode (c:\Users\me\.vscode\extensions\liwangqian.luacoderassist-2.2.7\server\lib\engine\core.js:467:17)
    at Array.forEach (<anonymous>)
[Info  - 11:05:08 PM] luacheck check file:///d%3A/test/test.lua in 35 ms.
```

**Comment by liwangqian** [comment]
Well, thanks for you feedback, this bug seems fixed in VER 2.2.8-prelease.

**Comment by grilledham** [comment]
Thanks, with 2.2.8 the issue is resolved for me.
